spec/requests/admin/nested_sidebar_spec.rb

```diff
@@ -42,10 +42,55 @@
     it "shows nested grandchildren items where applicable" do
       allow(FeatureFlag).to receive(:enabled?).with(:admin_restructure).and_return(true)
 
-      get admin_content_manager_badges_url
+      get admin_content_manager_badges_path
 
       expect(response.body).to include("Library")
       expect(response.body).to include("Achievements")
     end
   end
+
+  describe "profile admin feature flag" do
+    before do
+      allow(FeatureFlag).to receive(:enabled?).with(:admin_restructure).and_return(true)
+    end
+
+    it "does not show the option in the sidebar when the feature flag is disabled" do
+      allow(FeatureFlag).to receive(:enabled?).with(:profile_admin).and_return(false)
+
+      get admin_content_manager_articles_path
+
+      expect(response.body).not_to include("Profile Fields")
+    end
+
+    it "shows the option in the sidebar when the feature flag is enabled" do
+      allow(FeatureFlag).to receive(:enabled?).with(:profile_admin).and_return(true)
```

Comment:
@citizen428 I was wondering if you had any ideas on how I could get this test passing? The problem here is that in the tests when the feature flag is set, the AdminMenu class it not called again, hence its not recreating the payload and updating the `visible` attribute in the payload. If the `visible` attribute is not updated then the navbar item will not be shown.

Reply:
Sorry I missed this question 😢 I'm afraid I don't fully understand all the context here. Where does `AdminMenu` originally get called?